packages/react-button/src/components/MenuButton/useMenuButton.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { getSlots, resolveShorthandProps, mergeProps } from '@fluentui/react-compose/lib/next/index';
+import { MenuButtonProps, MenuButtonState } from './MenuButton.types';
+import { useMenuButtonState } from './useMenuButtonState';
+
+const menuButtonShorthandProps = ['icon', 'children', 'menuIcon', 'menu'];
+
+/**
+ * Redefine the render function to add slots. Reuse the button structure but add
+ * slots to children.
+ * @param state
+ */
+const renderMenuButton = (state: MenuButtonState) => {
+  const { slots, slotProps } = getSlots(state, menuButtonShorthandProps);
+  const { iconOnly, expanded } = state;
+
+  return (
+    <slots.root {...slotProps.root}>
+      <slots.icon {...slotProps.icon} />
+      {!iconOnly && <slots.children {...slotProps.children} />}
+      {!iconOnly && <slots.menuIcon {...slotProps.menuIcon} />}
```

Comment:
Should we render the `spans` even when these are not present?

Reply:
Similar for the other render functions.

Reply:
Ah, now you are seeing why I made the `getSlots` optimization to say: If the slot is a primitive, do not render it if children is `undefined`. It eliminates all of the redundant condition checks and makes things consistent.

I am starting to think it should be reverted because it causes magic.

Reply:
https://github.com/microsoft/fluentui/issues/14374